### DISCUSSION TITLE
Ability to source a provider resource?

### DISCUSSION THREAD

**Discussion by nwmcsween** [gruntwork-io/knowledge-base#466] [Feedback]
Basically I've seen most tutorials on terragrunt recommending to wrap a resource in a module but the issue with this is you end up re-implementing half of gcp, azurerm, etc with default args, ideally I could:
````
terraform {
  source = $PROVIDER/$RESOURCE
}
````
Or similar which would inherit vars, output, etc from the resource, this would remove about 3/4 of what currently makes up my terragrunt repo.

---

<ins datetime="2022-06-13T23:34:36Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108763">Tracked in ticket #108763</a></p>
</ins>

**Comment by yorinasub17** [comment]
Hi thanks for the feedback! I filed a feature request ticket for this feature in the `terragrunt` repo, here: https://github.com/gruntwork-io/terragrunt/issues/2142
